cupyx/scipy/ndimage/measurements.py

```diff
@@ -200,3 +203,297 @@ def _kernel_finalize():
+    sum, count = _ndimage_mean_kernel(input,
+                                      labels, index, index.size, out, count)
+    if return_count:
+        return sum/count, count
+    return sum/count
+
+
+def variance(input, labels=None, index=None):
+    """Calculate the variance of the values of an n-D image array, optionally
+    at specified sub-regions.
+
+    Args:
+        input (cupy.ndarray) :Nd-image data to process.
+        labels (cupy.ndarray or None): Labels defining sub-regions in `input`.
+            If not None, must be same shape as `input`.
+        index (cupy.ndarray or None): `labels` to include in output.  If None
+            (default), all values where `labels` is non-zero are used.
+
+    Returns:
+        variance (cupy.ndarray): Values of variance, for each sub-region if
+        `labels` and `index` are specified.
+
+
+    .. seealso:: :func:`scipy.ndimage.variance`
+    """
+    if not isinstance(input, cupy.ndarray):
+        raise TypeError('input must be cupy.ndarray')
+
+    if issubclass(input.dtype.type, (cupy.bool, cupy.complex64,
+                                     cupy.complex128)):
+        raise TypeError("cupyx.scipy.ndimage.variance doesn't support %{}"
+                        "".format(input.dtype.type))
+
+    use_kern = False
+    # There is constraints on types because atomicAdd() in CUDA 7.5
+    # only supports int32, uint32, uint64, and float32.
+    if not issubclass(input.dtype.type,
+                      (cupy.int32, cupy.float16, cupy.float32,
+                       cupy.float64, cupy.uint32, cupy.uint64,
+                       cupy.ulonglong)):
+        warnings.warn(
+            'Using the slower implmentation as '
+            'cupyx.scipy.ndimage.sum supports int32, float16, '
+            'float32, float64, uint32, uint64 as data types'
+            'for the fast implmentation', util.PerformanceWarning)
+        use_kern = True
+
+    if labels is None:
+        return input.var()
+
+    if not isinstance(labels, cupy.ndarray):
+        raise TypeError('label must be cupy.ndarray')
+
+    if index is None:
+        return (input[labels != 0]).var()
+
+    input, labels = cupy.broadcast_arrays(input, labels)
+
+    if not isinstance(index, cupy.ndarray):
+        if not isinstance(index, int):
+            raise TypeError('index must be cupy.ndarray or a scalar int')
+        else:
+            return (input[labels == index]).var()
+
+    mean_val, count = _mean_driver(input, labels, index, True, use_kern)
+    if use_kern:
+        new_axis = (..., *(cupy.newaxis for _ in range(input.ndim)))
+        return cupy.where(labels[None, ...] == index[new_axis],
+                          cupy.square(input - mean_val[new_axis]),
+                          0).sum(tuple(range(1, input.ndim + 1))) / count
+    out = cupy.zeros_like(index, dtype=mean_val.dtype)
+    return _ndimage_variance_kernel(input, labels, index, index.size, mean_val,
+                                    out)/count
```

Comment:
nit: Just for the style guideline.
```suggestion
                                    out) / count
```